### PR TITLE
Fix potential memory leak.

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
Fix of a cppcheck reported memory leak.

### CODE REVIEW

File: src/Engine/Manager/ScriptManager.cpp [hunk 1]
```diff
@@ -663,16 +663,13 @@ Component::Script* ScriptManager::CreateScript(const Json::Value& node) {
...
                 }
                 else if (typeId == asTYPEID_FLOAT){
-                    float* value = new float(typeId_value[typeIds[0]].asFloat());
-                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)value);
+                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)(new float(typeId_value[typeIds[0]].asFloat())));
                 }
                 else if (typeId == engine->GetTypeIdByDecl("string")){
-                    std::string* value = new std::string(typeId_value[typeIds[0]].asString());
-                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)value);
+                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)(new std::string(typeId_value[typeIds[0]].asString())));
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MAJOR Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)

File: src/Engine/Manager/ScriptManager.cpp [hunk 2]
```diff
@@ -663,16 +663,13 @@ Component::Script* ScriptManager::CreateScript(const Json::Value& node) {
                 std::vector<std::string> typeIds = typeId_value.getMemberNames();
                 int typeId = std::atoi(typeIds[0].c_str());
                 if (typeId == asTYPEID_INT32){
-                    int* value = new int(typeId_value[typeIds[0]].asInt());
-                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)value);
+                    script->propertyMap[name] = std::pair<int, void*>(typeId, (void*)(new int(typeId_value[typeIds[0]].asInt())));
                 }
                 else if (typeId == asTYPEID_FLOAT){
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/ScriptManager.cpp]: MAJOR Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/1c22a1201a3c8f752a4a6181b1a616aa011e802c/src/Engine/Manager/ScriptManager.cpp#L676): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L690](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/1c22a1201a3c8f752a4a6181b1a616aa011e802c/src/Engine/Manager/ScriptManager.cpp#L690): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [ScriptManager.cpp#L466](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/1c22a1201a3c8f752a4a6181b1a616aa011e802c/src/Engine/Manager/ScriptManager.cpp#L466): comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Chainsawkitten** [comment]
This does nothing to actually fix the leak, though?

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c1ce37509b6278c1280ef94768aa91169def2de6/src/Engine/Manager/ScriptManager.cpp#L676): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L690](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c1ce37509b6278c1280ef94768aa91169def2de6/src/Engine/Manager/ScriptManager.cpp#L690): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [ScriptManager.cpp#L466](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c1ce37509b6278c1280ef94768aa91169def2de6/src/Engine/Manager/ScriptManager.cpp#L466): comma should be followed by whitespace [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**